### PR TITLE
(#80) (#110) (#114) blazor storage implementation with partial friends service.

### PR DESCRIPTION
**Pull request by SaintAngeLs**
Dear Team,

I trust this message finds you well. I'm pleased to present our latest pull request (#115), which showcases significant advancements in our project, specifically focusing on implementing Blazor Storage alongside full integration of the Friends Service (#80, #110, #114).

This pull request encapsulates a total of ~ 72 commits, each contributing to the development of the Blazor Storage infrastructure and enhancements to the Friends Service frontend page. Here's a succinct overview of the key changes:

- Updates to service app settings and operating scripts to ensure optimal functionality (#80).
- Implementation of fundamental events for the service (#80).
- Integration of robust exceptions handling within the service layer (#80).
- Refinements and updates to the application layer to improve overall performance and usability (#80).
- Introduction of handlers for Friends Service events to facilitate seamless interactions (#82).
- Implementation of necessary endpoints for the Friends Service and addressing related bugs (#80).
- Integration of Blazor Storage for efficient data management and security enhancements (#80, #110).
- Implementation of features such as profile connections, notifications, and more to enrich user interaction (#80).

Additionally, this pull request encompasses resolution of conflicts, merges with the test branch, and comprehensive test coverage to maintain code quality.

Kindly note the two unresolved conversations linked to the rendering updates in the [MainLayout.razor](link-to-conversation) file. These issues are earmarked for resolution in subsequent iterations.

Your insights and feedback on the changes proposed are highly valued as we strive for continuous improvement and excellence in our project. Please take the time to review the modifications and share your thoughts.

Thank you for your attention to this pull request. We look forward to your feedback and eventual merge.

### CODE REVIEW

File: MiniSpace.Web/src/MiniSpace.Web/Shared/MainLayout.razor [hunk 1]
```diff
@@ -5,12 +5,19 @@
...
             <RadzenStack Orientation="Orientation.Horizontal" AlignItems="AlignItems.Center" Gap="0">
-                @if (IdentityService.IsAuthenticated && StudentsService.StudentDto.State == "valid")
+                @* 
+                    TODO: do not forget about the code snippet chaned ❕❕❕
+                        @if (IsUserAuthenticated && StudentsService.StudentDto.State == "valid") 
+                *@
+                @if (IsUserAuthenticated) 
                 {
                     <RadzenSidebarToggle Click="@(() => _sidebarExpanded = !_sidebarExpanded)" />
                 }
```
Comment (SaintAngeLs) [inline MiniSpace.Web/src/MiniSpace.Web/Shared/MainLayout.razor:23]: Currently I will not resolve this one. Please do not forget about that in the next iteration. This more like 'carving up' with the identity service and the local storage with the students service.

File: MiniSpace.Web/src/MiniSpace.Web/Shared/MainLayout.razor [hunk 2]
```diff
@@ -49,12 +60,14 @@
                                          Click="@(() => NavigationManager.NavigateTo("account"))" />
                     <RadzenPanelMenuItem Text="Events" Icon="event"
                                          Click="@(() => NavigationManager.NavigateTo("events"))" />
+                    <RadzenPanelMenuItem Text="Friends" Icon="group"
+                             Click="@(() => NavigationManager.NavigateTo("friends"))" />
```
Comment (SaintAngeLs) [inline MiniSpace.Web/src/MiniSpace.Web/Shared/MainLayout.razor:64]: This is technically-practically the issue related to previous one:

Currently I will not resolve this one. Please do not forget about that in the next iteration. This more like 'carving up' with the identity service and the local storage with the students service.

**Comment by an2508374** [comment]
Generally, this changes seem to be a great improvement.

But firstly, I've observed that directly after signing in the authentication state is not updated, it is needed to go to home page and then we can see sidebar which means that user is authenticated.

And secondly, please pay attention to the fact that going to the events page sometimes can be associated with throwing some exception, it is connected with changes provided in #110 and will be fixed by me in one or two days by upcoming pull request from frontend_organizations branch.

**Review by an2508374 — APPROVED** [review]

**Comment by SaintAngeLs** [comment]
> Generally, this changes seem to be a great improvement.
> 
> But firstly, I've observed that directly after signing in the authentication state is not updated, it is needed to go to home page and then we can see sidebar which means that user is authenticated.
> 
> And secondly, please pay attention to the fact that going to the events page sometimes can be associated with throwing some exception, it is connected with changes provided in #110 and may be fixed by me in one or two days after upcoming pull request from frontend_organizations branch.

Yeap. you are totally correct. After signing in currently there is a page reload needed. I will handle it in one of the next pr.

About the changed provided in frontend_organisations branch, lets see how it goes. I will strongly believe, the updated with this pr will not provide conflicts to resolve.

**Pull request merged**